taxi/tests/test_views.py

```diff
@@ -0,0 +1,84 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car, Driver
+from taxi.views import ManufacturerListView
+
+
+def create_new_user_setup(obj):
+    obj.user = get_user_model().objects.create_user(
+        username="test_username", password="test_password"
+    )
+
+    obj.client.force_login(obj.user)
+
+
+class PublicTest(TestCase):
+    def test_login_required_for_main_pages(self):
+        self.assertNotEqual(self.client.get(reverse("taxi:car-list")), 200)
+        self.assertNotEqual(
+            self.client.get(reverse("taxi:manufacturer-list")), 200
+        )
+        self.assertNotEqual(self.client.get(reverse("taxi:driver-list")), 200)
+        self.assertNotEqual(self.client.get(reverse("taxi:index")), 200)
```

Comment:
Don't test several pages like that. Use subtests or write a separate test for each view.
Chaining assertions results into only one meaningful test result (as opposed to meaningful results or every test) when something fails.

Also, I think it would be justified to use `.assertRedirects()` here.

Reply:
yes, @AVyha just make it in different tests